### PR TITLE
Fix(#4106) improve responsiveness of profile page

### LINKED ISSUE #4106: improve responsiveness of profile page
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

As you can see in the image, The My Circuits | Favourite Circuits | Collaborated Circuits and the Search bar are collapsing together and making it not responsive in the device width.

### Steps To Reproduce

1. Go to the main website [[circuitverse.org](https://circuitverse.org/)]
2. Log in yourself
3. Open your profile in mobile

### Expected Behavior

We need to make it a bit more responsive

In desktop mode, the layout will be
```
-----------------------------------------------------------------------------------------------
| My Circuits | Favourite Circuits | Collaborated Circuits | <-----------------> | Search Bar | 
-----------------------------------------------------------------------------------------------
```

In mobile mode, the layout will be like (based on the device  width)
```
------------------------------------------------------------
| My Circuits | Favourite Circuits | Collaborated Circuits | 
------------------------------------------------------------
|                      Search Bar                          | 
------------------------------------------------------------
```

Or

```
------------------------------------------------------------
|          My Circuits       |      Favourite Circuits     |
------------------------------------------------------------
|   Collaborated Circuits    | 
------------------------------------------------------------
|                      Search Bar                          | 
------------------------------------------------------------
```

Make sure the text of each button come in single line

### Are you working on this issue?

No

### PR DESCRIPTION
**Pull request by hrshkshri**
Fixes #4106 

#### Describe the changes you have made in this PR -

1. Responsive User circuit navbar
2. Full-width search bar

### Screenshots of the changes (If any) -

**Before**
Screenshot 2023-10-05 at 1 20 18 PM

Screenshot 2023-10-05 at 1 21 49 PM

**After**
Screenshot 2023-10-16 at 1 04 37 AM

Screenshot 2023-10-16 at 1 04 50 AM

### CODE REVIEW

**Comment by hrshkshri** [comment]
Hey @tachyons @VaibhavUpreti @vedant-jain03, I made some changes that needed a review here.

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]
Let's just make a change in mobile design only, the intial web design looks good to me.

**Comment by hrshkshri** [comment]
> Let's just make a change in mobile design only, the intial web design looks good to me.

Hey @vedant-jain03 new change looks like this now 

**For web**

 
Screenshot 2023-10-06 at 10 30 15 AM

**For small devices <766px**

Screenshot 2023-10-06 at 10 32 03 AM

File: app/assets/stylesheets/users.scss [hunk 1]
```diff
@@ -351,12 +359,13 @@
...
 .users-circuits-tab {
   background-color: $form-border-grey;
-  border-radius: 0;
+  border-radius: 5px;
   color: $black;
   font-size: 18px;
-  padding: 7px 25px;
-  margin: 0;
+  margin-left: 7px;
+  margin-right: 7px;
```
Comment (tanmoysrt) [inline app/assets/stylesheets/users.scss:366]: For gap, avoid adding margin directly to button.
Instead, add `gap` in the parent class `users-circuits-navbar-buttons`

**Review by tanmoysrt — CHANGES_REQUESTED** [review]

**Comment by tanmoysrt** [comment]
@hrshkshri just a little change if you can add

For the search bar in mobile screen, add border to all side of the input box. For desktop keep the current one, just left border
Reference class `.users-circuits-navbar input`

File: app/assets/stylesheets/users.scss (hunk 1 above)
Comment (hrshkshri) [reply app/assets/stylesheets/users.scss]: Done sir.

File: app/assets/stylesheets/users.scss [hunk 2]
```diff
@@ -313,6 +313,7 @@
 .users-circuits-navbar {
   align-items: center;
   display: flex;
+  // flex-direction: column;
```
[the remark this answers is not in the document]
Comment (hrshkshri) [reply app/assets/stylesheets/users.scss]: Yeah did this too

File: app/assets/stylesheets/users.scss [hunk 3]
```diff
@@ -330,16 +330,26 @@
     }
   }
 }
+
+.users-circuits-navbar-buttons {
+  align-items: center;
+  display: flex;
+  justify-content: space-between;
+  gap: 10px;
```
[the remark this answers is not in the document]
Comment (tanmoysrt) [reply app/assets/stylesheets/users.scss:337]: @hrshkshri I think we can just specify `column-gap` instead of `gap`

**Comment by tanmoysrt** [comment]
@hrshkshri the changes look fine, resolve codeclimate issues!

**Comment by hrshkshri** [comment]
> @hrshkshri the changes look fine, resolve codeclimate issues!

What about `gap` property code climate says unknown property

**Comment by tanmoysrt** [comment]
> > @hrshkshri the changes look fine, resolve codeclimate issues!
> 
> What about `gap` property code climate says unknown property

Use `column-gap`, I think `gap` has been deprecated as latest standard

**Review by tanmoysrt — APPROVED** [review]
LGTM :+1:

**Comment by tanmoysrt** [comment]
Hi @hrshkshri just one thing, update the screenshots of issue description

**Comment by hrshkshri** [comment]
> Hi @hrshkshri just one thing, update the screenshots of issue description


https://github.com/CircuitVerse/CircuitVerse/assets/108923011/9fa89adc-0de9-43ab-bde9-53bbfdb1e851

**Comment by hrshkshri** [comment]
Hey @vedant-jain03, Sir needed your review here... : )

**Comment by tanmoysrt** [comment]
> > Hi @hrshkshri just one thing, update the screenshots of issue description
> 
>  demo.mov

put this in description

**Comment by hrshkshri** [comment]
> > > Hi @hrshkshri just one thing, update the screenshots of issue description
> > 
> > 
> > demo.mov
> 
> put this in description

Sorry, I don't understand... What else I need to do?

**Comment by niladrix719** [comment]
@hrshkshri put the updated video in the issue description

**Comment by hrshkshri** [comment]
Hey @vedant-jain03, All requested changes are done here... waiting for your review  here : )

**Review by vedant-jain03 — APPROVED** [review]

**Review by tanmoysrt — APPROVED** [review]

**Pull request merged**